scripts/metalk8s_cli/solutions/config.py

```diff
@@ -0,0 +1,123 @@
+"""Helper module for managing a SolutionsConfiguration file."""
+import argparse
+import os.path
+
+import yaml
+
+from metalk8s_cli.exceptions import CommandInitError, CommandError
+
+
+class SolutionsConfiguration(object):
+    KIND = 'SolutionsConfiguration'
+
+    # NOTE: first apiVersion in this list is used when creating a new file
+    API_VERSIONS = [
+        'solutions.metalk8s.scality.com/{}'.format(version)
+        for version in ['v1alpha1']
+    ]
+
+    def __init__(self, data, filepath=None):
+        api_version = data.get('apiVersion')
+        kind = data.get('kind')
+        assert api_version and kind, (
+            'SolutionsConfiguration must have a `kind` and `apiVersion`.'
+        )
+        assert kind == self.KIND, (
+            'Wrong kind ({}), must be "{}".'.format(kind, self.KIND)
+        )
+        assert api_version in self.API_VERSIONS, (
+            'API version not supported: {}'.format(api_version)
+        )
```

Comment:
Hmm you do the same check as in the salt module. That mean if for what ever reason we need to change something we need to change it in salt and in the script. Maybe you should use a salt module to check this config file.

Reply:
Good point. Given it's a simple `salt-call --local`, I guess the performance penalty of delegating this to Salt is acceptable (this CLI is already slow as hell due to the numerous calls to Salt master - might be good to profile a bit more what takes time, I fear having expensive ext_pillars is not that good for responsiveness of the system).

Reply:
Also, since Salt is responsible for defining format/usage of this config file (same as it would be for `/etc/metalk8s/bootstrap.yaml` I guess), I think it should also be responsible for managing its path (for now, it's hardcoded in `__salt__['metalk8s_solutions.SOLUTIONS_CONFIG_FILE']`). Which means I should remove the `--solutions-config` argument.

Reply:
Done, see 65d8738